Description: Use ld.bfd explicitly
 Freepascal is broken with ld.gold, previously we conflicted with binutils-gold
 but that makes us uninstallable with the new binutils.
 
 So instead we take the approach of patching fpc to use ld.bfd directly
Author: Peter Michael Green <plugwash@debian.org>
Bug-Debian: http://bugs.debian.org/620815
Bug-Debian: http://bugs.debian.org/624525
Bug-Debian: http://bugs.debian.org/717651

Index: fpc/fpcsrc/compiler/systems/t_linux.pas
===================================================================
--- fpc.orig/fpcsrc/compiler/systems/t_linux.pas
+++ fpc/fpcsrc/compiler/systems/t_linux.pas
@@ -1408,7 +1408,7 @@ begin
   if HasExports then
     cmdstr:=cmdstr+' -E';
 
-  success:=DoExec(FindUtil(utilsprefix+BinStr),CmdStr,true,false);
+  success:=DoExec(FindUtil(utilsprefix+BinStr)+'.bfd',CmdStr,true,false);
 
   { Create external .dbg file with debuginfo }
   if success and (cs_link_separate_dbg_file in current_settings.globalswitches) then
@@ -1476,7 +1476,7 @@ begin
   Replace(cmdstr,'$SONAME',SoNameStr);
   Replace(cmdstr,'$MAP',mapstr);
   Replace(cmdstr,'$GCSECTIONS',GCSectionsStr);
-  success:=DoExec(FindUtil(utilsprefix+binstr),cmdstr,true,false);
+  success:=DoExec(FindUtil(utilsprefix+binstr)+'.bfd',cmdstr,true,false);
 
 { Strip the library ? }
   if success and (cs_link_strip in current_settings.globalswitches) then
